src/javascripts/components/navigation.js

```diff
@@ -0,0 +1,144 @@
+import 'govuk-frontend/govuk/vendor/polyfills/Element/prototype/classList'
+
+import common from 'govuk-frontend/govuk/common'
+var nodeListForEach = common.nodeListForEach
+
+var navActiveClass = 'app-navigation--active'
+var navMenuButtonActiveClass = 'govuk-header__menu-button--open'
+var subNavActiveClass = 'app-navigation__subnav--active'
+// This one has the query dot at the beginning because it's only ever used in querySelector calls
+var subNavJSClass = '.js-app-navigation__subnav'
+
+function Navigation ($module) {
+  this.$module = $module || document
+
+  this.$nav = this.$module.querySelector('.js-app-navigation')
+  this.$navToggler = this.$module.querySelector('.js-app-navigation__toggler')
+  this.$navButtons = this.$module.querySelectorAll('.js-app-navigation__button')
+  this.$navLinks = this.$module.querySelectorAll('.js-app-navigation__link')
+
+  // Create variable that saves the last recorded state window size for use by the saveStates function
+  this.savedWindowWidth = null
+}
+
+// Helper function to check if the screen size is above or below the tablet viewport cutoff
+Navigation.prototype.windowSize = function () {
+  return document.documentElement.clientWidth >= 641 ? 'big' : 'small'
+}
+
+// Checks if the saved window size has changed between now and when it was last recorded (on load and on viewport width changes)
+// Reapplies hidden attributes based on if the viewport has changed from big to small or vice verca
+// Saves the new window size
+Navigation.prototype.setHiddenStates = function () {
```

Comment:
I'd recommend taking a look at [`window.matchMedia`](https://developer.mozilla.org/en-US/docs/Web/API/Window/matchMedia) as it provides a native way to listen for breakpoint changes – there's [an example of it in action with the tabs component in GOV.UK Frontend](https://github.com/alphagov/govuk-frontend/blob/main/src/govuk/components/tabs/tabs.js).

Reply:
TIL! Do we reckon that it's reasonable the responsive movement be a progressive enhancement for users on IE9 and below? So the responsive updating would only apply as a progressive enhancement. My instinct is that this is fine, otherwise I'd need to provide 2 ways to check for breakpoint changes. This also feels correct with the impending browser support work.